zebra/zebra_vty.c

```diff
@@ -1793,6 +1793,21 @@ DEFUN (show_evpn_vni,
 	return CMD_SUCCESS;
 }
 
+DEFUN(show_evpn_vni_detail, show_evpn_vni_detail_cmd,
```

Comment:
Shouldn't this be a "show evpn vni all" command instead of "detail". 

1) The information is not more than what it would have displayed if a specific VNI is mentioned. It is showing the information for "all" VNIs. 
2) EVPN already has such commands with "all" Key word.  So "all option would be better than "detail".

Reply:
Nitin,

As per the commit message, this command is to print multiple/All EVPN entries. It has detail option but not "all" option. I didn't see any "all" option in the new code as well. So, I am not clear on your response. 

Let me know in case my comment is not clear.

-Mohan

Reply:
@srimohans My bad. I got confused with other "detail" options I am adding as part of CLIs -
net show evpn mac vni <all|id> [detail]
net show evpn arp-cache vni <all|id> [detail]

I will be pushing the code for above two CLIs soon. The idea behind adding "show evpn vni detail" is to make this consistent with the above two CLIs. As you can see that above 2 CLIs have an "all" option which prints the tabular summary of all possible VNIs. A "detail" option will be printing the detail information for all possible VNIs.

We probably might have an "all" option ("show evpn vni all") later in the future just to print the tabular summary hence IMO "detail" would be a better choice. Do let me know if you feel otherwise.

Sorry for the confusion.

Reply:
Turns out, we print summary using "show evpn vni" hence using "detail" would be just to keep it consistent with the above mentioned CLIs

```
bordertor-11# show evpn vni
VNI        Type VxLAN IF              # MACs   # ARPs   # Remote VTEPs  Tenant VRF
1000114    L2   vxlan1000114          3        9        0               vrf1
1000112    L2   vxlan1000112          5        15       2               vrf1
1000113    L2   vxlan1000113          3        9        0               vrf2
1000111    L2   vxlan1000111          5        15       2               vrf2
104001     L3   vxlan104001           2        4        n/a             vrf1
104002     L3   vxlan104002           2        4        n/a             vrf2

```

Reply:
@srimohans good question. I did expect some discussion around the choice of keyword here as we need to set the right precedent for all FRR display commands. As @nitinsoniism explained "all" is already taken for other commands even if not for this specific one. The way I see it -
1. "all" or the absence of a specific id implies - show "brief" output for all entries in a database; typically in a tabular format.
2. we need a new key word for displaying "detailed" output. currently when an <id> is specified detailed output is dumped for the specific id. however we found that detailed info is needed for debugging on "live" PODs. Asking customers to intelligently gather info is a total PITA (find the id first and then use it etc.). So we do need "detailed" output without specifying <id>. After much debate we arrived on "detail". Other routers (for e.g. Cisco) also use the "detail" keyword for similar purposes as well - `cisco-n9k-1# sh nve interface nve 1 ?
  <CR>    
  >       Redirect it to a file
  >>      Redirect it to a file in append mode
  detail  Detailed information
  |       Pipe command output to filter

cisco-n9k-1# sh nve interface nve 1  `

Reply:
@AnuradhaKaruppiah 

I understand the need to collect the list of all VNIs, instead of specifying a particular VNI through ID. 

I am little confused after going through the updates from you and Nitin.  

1) As per Nitin's last update, "show evpn vni" already prints all the VNIs as shown above. 
2) Also, as part of this new option for the command, no "Detailed" output is being printed, but it is still a "brief" output of all VNIs. Am I right ?

If so, why do we need to add this option to this command, as the requested output is already present without this option as well.

Reply:
@AnuradhaKaruppiah 

This would be a combination of all & detail, but I think just having "detail" key word for this output is also good. 

Thanks for the explanation. Initially, I was little confused with the updates. 

Will approve.

Reply:
awesome. thx.